src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraEntityBsonSupport.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.agora.server.model
+
+import com.mongodb.DBObject
+import org.bson.types.ObjectId
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+
+
```

Comment:
another architectural option: create a wrapper around DBObject, a  la:
```
class DBObjectWrapper(obj: DBObject) {
  def getOrNone[A](key: String): Option[A] =
    Option(obj.get(key)).map(_.asInstanceOf[A])

  def entityType: Option[AgoraEntityType.EntityType] =
    getOrNone[String]("entityType").map(AgoraEntityType.withName)
}
```

then your  `read` method becomes more terse because of OO:
```
    val dbo = new DBObjectWrapper(obj)
    AgoraEntity(
      namespace       = dbo.getOrNone("namespace"),
      name            = dbo.getOrNone("name"),
      ...
```

I defer to you on whether or not you prefer this approach, and I defer to the profiler on whether or not it's better to create a new wrapper object for each DBObject or if it's better to continually pass the DBObject into methods.

Reply:
Isn't passing it into methods basically free because it's happening by reference?